src/read_restart.cpp

```diff
@@ -207,7 +207,24 @@ void ReadRestart::command(int narg, char **arg)
     memory->create(buf,assignedChunkSize,"read_restart:buf");
     mpiio->read((headerOffset+assignedChunkOffset),assignedChunkSize,buf);
     mpiio->close();
-
+    if (!nextra) { // We can actually calculate number of atoms from assignedChunkSize
```

Comment:
I'll have to see if there is a cleaner way of determining if the per-atom data is a constant size, and on how to get that size.  This seems a bit of a hack...
Also, as we discovered, on very large MPI rank counts, it might not be best for the IO system to have this evenly divide up the work of reading the restart file... it might be better to consolidate to fewer MPI ranks.  Though, that is what MPI-IO is supposed to do behind the scene itself... so, we (I) may need to try a few variants, and benchmark them.